### PR TITLE
SS-2298 Initialize db schemas

### PR DESCRIPTION
**Pull request by frankban**
When the database connection information becomes available from the temporal char, set up database schemas for both visibility and temporal databases and then notify that the schema is ready back to the temporal units.

I also moved forward with the implementation of the `tctl` action, but I don't expect that to be working yet.

### CODE REVIEW

**Review by TeodorPt — APPROVED** [review]

**Review by kian99 — APPROVED** [review]
Very nice!

File: src/charm.py [hunk 1]
```diff
@@ -100,12 +103,82 @@ def _setup_db_schemas(self, event):
...
+                "--port",
+                database_connection["port"],
+                "--database",
+                database_connection["dbname"],
+                "--user",
+                database_connection["user"],
+                "--password",
+                database_connection["password"],
+                "setup-schema",
+                "-v",
```
[the remark this answers is not in the document]
Comment (frankban) [reply src/charm.py:128]: I don't know yet.

File: src/charm.py [hunk 2]
```diff
@@ -76,15 +75,19 @@ def _on_admin_relation_changed(self, event):
...
+
         args = event.params["args"].split()
-        cmd = ["tctl"] + args
-        event.log(f"running this tctl command: {' '.join(cmd)}")
-        proc = subprocess.run(cmd, capture_output=True)
-        if proc.returncode != 0:
-            err = proc.stderr.decode("utf-8")
-            event.fail(f"command failed with code {proc.returncode}\n{err}")
+        try:
+            output = exec(container, "tctl", *args)
```
[the remark this answers is not in the document]
Comment (frankban) [reply src/charm.py:85]: Good point, I'll do.

File: src/charm.py [hunk 3]
```diff
@@ -100,12 +103,82 @@ def _setup_db_schemas(self, event):
...
+
+        admin_relations = self.model.relations["admin"]
+        if not admin_relations:
+            # Can this happen? Probably in a race between hook execution and
+            # removed relation?
+            logger.debug("admin:temporal: not notifying schema readiness: admin relation not available")
+            self.unit.status = BlockedStatus("admin:temporal relation: not available")
+            return
+        logger.info("notifying schemas are ready")
+        for relation in admin_relations:
```
[the remark this answers is not in the document]
Comment (frankban) [reply src/charm.py:159]: There might be multiple temporal servers connected to the admin charm, and all of them need to be notified that they can start.

**Pull request merged**